### DISCUSSION TITLE
Writing back editable cells

### DISCUSSION THREAD

**Discussion by theref** [anvilistas/tabulator#42] [Q&A]
In the code for the example apps you mention that you can **edit the data inplace if writeback is set to True**

```
   def tabulator_cell_edited(self, cell, **event_args):
        """This method is called when a row is edited - returns the row - will edit the data inplace if writeback is set to True"""
        print(f"{cell.get_field()} from row: {cell.get_row().get_index()} edited")
        print(f"previously: {cell.get_old_value()!r} now {cell.get_value()!r}")
        self.tabulator_row_formatter(cell.get_row())
```

at what point do I set `writeback=True`?

I'm not sure if it makes a difference, but I'm using a list of models rather than a list of dicts. with:
```
            use_model=True,
            getter= getattr,
```

**Comment by s-cork** [comment]
I'm not really sure. That's statement is quite old so probably no longer relevant. 

I imagine it was talking about using a custom form as the edit component with writebacks. But future me isn't sure what past me meant 🤷‍♂️   

If you do edit the cell it will edit the value from row.get_data() which is a JavaScript proxy object of data. 

You'd need to write those changes back to your model I guess in this callback.

**Reply by theref** [reply]
ok, so something along the lines of:

```
def tabulator_cell_edited(self, cell, **event_args):
        cell.get_model().ATTRIBUTE = cell.get_value()
```

**Reply by s-cork** [reply]
Yeah exactly.

**Reply by theref** [reply]
thank you!
